src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs

```diff
@@ -769,11 +769,12 @@ public static FormulaValue ModFloat(IRContext irContext, NumberValue arg0nv, Num
 
             double result = arg0 - (arg1 * ((long)q));
 
-            // We validate the reminder is in a valid range.
-            // This is mainly to support very large numbers (like 1E+308) where the calculation could be incorrect
-            if (result < -Math.Abs(arg1) || result > Math.Abs(arg1))
+            // If result is overflow or (arg0 is more than 1e15 and result = 0 - highly wrong result), using C# mod function
+            if ((Math.Abs(arg0) > 1e15 && result == 0) || (result < -Math.Abs(arg1) || result > Math.Abs(arg1)))
             {
-                return CommonErrors.OverflowError(irContext);
+                // C# % result has the same sign with dividend.
+                // Using the following fomular to have the sign of mod result is the same as divisor as in Excel and PA Mod
+                result = (arg0 % arg1) * Math.Sign(arg0) * Math.Sign(arg1);

```

Comment:
This is wrong - instead of flipping the signal, we should add the divisor (if the result is negative and should be positive) or subtract the divisor (if the result is positive and should be negative)